Adds partial Command integration for Non-Humans

Anvilman6 opened the pull request:
Allows Non-Humans to enroll in specific Command positions, dependent on species, with heavy consideration given to lore and code-based strengths and weaknesses - developed with input and approval from the Yogstation Council. Applies to both roundstart and latejoin. Will be in **testmerge** format initially to allow for community feedback and bug hunting.

# General Documentation

### Intent of your Pull Request

To add carefully limited Command integration for non-Human species; thereby increasing round diversity, RP opportunities, and opening the path for more individuals to play as Head roles without being autobalanced to Human.

_NOTE: Non-Humans are still blacklisted from direct Captaincy and being Head of Personnel at roundstart or by latejoin._ 
_Invalid Non-Human / Command position combinations will still default to Human (I.E. Plasmaman Head of Security -> Human Head of Security)._

**Permitted Command position / Non-Human species combinations are as follows**:

**Head of Security**: Lizards
**Chief Engineer**: Plasmamen, Moths, Ethereals, Preternis, Polysmorphs
**Research Director**: Pods, Plasmamen, Ethereals, Preternis, Polysmorphs
**Chief Medical Officer**: Lizards, Pods, Moths
* Note: Pods are the code reference for Phytosians / podpeople (as listed on the Wiki)

### Why is this change good for the game?

This PR creates a small paradigm shift for Yogs, breaking the stale mold of seeing only Humans in Command roles in rounds at start or through latejoin. It also serves to create interesting power dynamics for RP situations, as there are players who already look down on Non-Humans, and said Non-Human being in a departmental head position can create interesting tension and conflict resolution. Additionally, a Command team of multiple Non-Humans paired with Humans also represents a more diverse and thereby more powerful workforce, able to tap into differing areas of strengths.

# Wiki Documentation

### What will the Wiki do to adjust for this change?

TODO: Relevant changes to each Playable Race page and the overall Jobs page (immediately), and retroactively adjusting references scattered about the Wiki (post testmerge). Wiki Team members will be tasked appropriately.

### What should players be aware of when it comes to the changes your PR is implementing?

1. That the Plasmaman wearing the RD's labcoat may not, in fact, be an imposter among us who killed the RD and stole his swag, and neither is that Podperson CMO! Diversity could potentially be more visible in departments, and racist comments may well get you a stern telebaton to the head.
2. That the AI **on Asimov** is still not _required_ to listen to Non-Humans, even if they are in Command.

### What general grouping does this PR fall under? 

Playable Races, Command pages, Jobs

# Changelog

:cl:  
rscadd: New Non-Human species specific Command position enrollment!
rscadd: A new entry in the game_options.txt which allows quick adding and subtracting of allowed Species / Command position combinations (JOB_SPECIES_WHITELIST)!
wip: If this PR survives the testmerge period and goes to fullmerge, other modular additions will be looked into, such as Command envirosuits for CE/RD Plasmapeople! This was deferred to prevent tying up of valuable Spriter time and resources.
experimental: If you find anything working not as intended with this testmerge active, or you'd like to give your thoughts / feedback on the PR, please say something on the Yogstation Discord to me at Gavius#3817 or leave a comment here!
/:cl:

Anvilman6 committed:
Adds partial Command integration for Non-Humans

Allows Non-Humans to enroll in Command positions, dependent on Species. Applies to both roundstart and latejoin.

alexkar598 reviewed (commented):
Hm yes code, very good code, I have never seen this code before so this is very impartial review mmmm code

Anvilman6 committed:
Ethereals, Polys moved from HoS to RD, CE 

Per lengthy consultation with other developers, I have decided to move Ethereals and Polysmorphs out of the HoS qualification into RD/CE respectively.
Main documentation to be adjusted accordingly, and fuller explanatory comment to be posted to thread shortly.

Anvilman6 committed:
Merge branch 'master' into SPECIES
